### ISSUE TITLE
The Oath addon (Amber) hanging on results

### ISSUE THREAD

**Issue by tapwizard** [bartolomesoriano/skin.amber#22]
Hi, the issue is while scraping for an episode in TV shows, after the scrape is done it hangs on loading for an extremely long time (minutes), It doesn't happen all the time (sometimes results are given immediately, but most times not). Movies seem to be ok, not 100% sure though. No errors thrown just takes extremely long to bring up results. This is happening on the current most recent version of Amber. I'm guessing it's a skin issue because I have tried The Oath on Estuary and Xonfluence skin and the hanging does not happen on those skins.

**Comment by bartolomesoriano** [comment]
@tapwizard , I am not sure that I can help with this issue, as I do not use any streaming addons other than Netflix and Disney+, and those are working fine.  I can take a look at a debug log of when you experience the issue, but I cannot promise anything.

**Comment by tapwizard** [comment]
I was looking at this issue a little further and it seems to be prevalent across addons that are code based on exodus. I have replicated the issue using exodus redux, scrubs v2, and the oath (I believe this has code based on exodus??).  I'll try to get a debug log. It is the same using your version or the version in kodi repo and even on Matrix. Thanks

**Comment by tapwizard** [comment]
log while doing multiple scrapes across various addons.
[kodi.log](https://github.com/bartolomesoriano/skin.amber/files/6047098/kodi.log)

**Comment by bartolomesoriano** [comment]
@tapwizard , I did not see any errors in the log that might point to what the issue is.  The only other thing I can think of is that some of the addons that Amber loads, such as favorites or skinshortcuts might be interfering with the  addon you are using.  Can you try with Aeon Nox Silvo or Arctic Horizon and let me know if your addon runs fine with those skins? They are also based on skinshortcuts.  Xonfluence and Estuary are not.

**Comment by tapwizard** [comment]
I have tried Arctic Horizon and seems to work fine. I checked version of skin shortcuts and they were different. Amber has 1.0.17 while Horizon has 1.0.19. I replaced Amber version with the one in Horizon and got some weird results. The hanging is still present  but if I move the mouse the results come up. Enabling or disabling mouse toggle in skin does nothing and if I disable mouse in kodi settings then it just hangs. I guess I'll just use addons that work with Amber as I love the Netflix style and putting favourites on a shelf where it shows what you have in favourites (only skin I know that does this). Thanks anyway just a very strange issue.
